src/ContentLanguageSwitcherProvider.php

```diff
@@ -62,8 +62,7 @@ public function __construct(LanguageManagerInterface $language_manager, PathMatc
    *   Array of available translation links.
    */
   public function getAvailableEntityLanguages(EntityInterface $entity) {
-    $route_name = $this->pathMatcher->isFrontPage() ? '<front>' : '<current>';
-    $links = $this->languageManager->getLanguageSwitchLinks(LanguageInterface::TYPE_CONTENT, Url::fromRoute($route_name));
+    $links = $this->languageManager->getLanguageSwitchLinks(LanguageInterface::TYPE_CONTENT, Url::fromRouteMatch(\Drupal::routeMatch()));
```

Comment:
In order to expedite things, this can stay as it is for now. However, the path matcher is no longer needed in this class and you should inject the route match. So you should open a followup to fix it.